0.10/README.md

```diff
@@ -90,18 +90,52 @@ file inside your source code repository.
+This image supports development mode. This mode can be switched on and off with the environment variable `DEV_MODE`. `DEV_MODE` can either be set to `true` or `false`.
+Development mode supports two features:
+* Hot Deploy
+* Debugging
+
+The debug port can be speicifed with the environment variable `DEBUG_PORT`. `DEBUG_PORT` is only valid if `DEV_MODE=true`.
+
+A simple example command for running the docker container in production mode is:
+```
+docker run --env DEV_MODE=true my-image-id
+```
+
+To run the container in development mode with a debug port of 5454, run:
+```
+$ docker run --env DEV_MODE=true DEBUG_PORT=5454 my-image-id
+```
+
+To run the container in production mode, run:
+```
+$ docker run --env DEV_MODE=false my-image-id
+```
+
+By default, `DEV_MODE` is set to `true`, and `DEBUG_PORT` is set to `5858`.
+
+Hot deploy
+--------------------
 
-* [Supervisor](https://github.com/petruisfan/node-supervisor)
-* [Nodemon](https://github.com/remy/nodemon)
+As part of development mode, this image supports hot deploy. If development mode is enabled, any souce code that is changed in the running container will be immediately reflected in the running nodejs application.
 
-Please note that in order to be able to run your application in development mode, you need to modify the [S2I run script](https://github.com/openshift/source-to-image#anatomy-of-a-builder-image), so the web server is launched by the chosen module, which checks for changes in the source code.
+### Using Docker's exec
 
-To change your source code in running container, use Docker's [exec](http://docker.io) command:
+To change your source code in a running container, use Docker's [exec](http://docker.io) command:
 ```
-docker exec -it <CONTAINER_ID> /bin/bash
+$ docker exec -it <CONTAINER_ID> /bin/bash
 ```
 
-After you [Docker exec](http://docker.io) into the running container, your current directory is set to `/opt/app-root/src`, where the source code is located.
+After you [Docker exec](http://docker.io) into the running container, your current directory is set to `/opt/app-root/src`, where the source code for your application is located.
+
+### Using OpenShift's rsync
+
+If you have deployed the container to OpenShift, you can use [oc rsync](https://docs.openshift.org/latest/dev_guide/copy_files_to_container.html) to copy local files to a remote container running in an OpenShift pod. Examples of integrating this behaviour with Grunt and Gulp's _watch_ tasks can be found [here](./test/test-app).
+
+#### Warning:
+Hot deploy is implemented with the default behaviour of [nodemon](https://github.com/remy/nodemon). The default behaviour is to execute the main attribute of the _package.json_ file, and failing that, to execute the start script. In production mode, sti-nodejs will execute the start script via `npm start -d`. It is advised to remove the _main_ attribute in the _package.json_ file when using hot deploy/development mode.
```

Comment:
```json
{
  "name": "node-echo",
  "version": "0.0.1",
  "description": "node-echo",
  "main": "server.js", <--- main attribute
  "dependencies": {
  },
  "devDependencies": {
    "nodemon": "*"
  },
  "engine": {
    "node": "*",
    "npm": "*"
  },
  "scripts": {
    "dev": "nodemon --ignore node_modules/ server.js",
    "start": "node server.js" <-- start script
  },
  "repository": {
    "type": "git",
    "url": "http://github.com/bettiolo/node-echo.git"
  },
  "keywords": [
    "Echo"
  ],
  "author": "Marco Bettiolo <marco@bettiolo.it>",
  "license": "",
  "bugs": {
    "url": "http://github.com/bettiolo/node-echo/issues"
  },
  "homepage": "http://apilb.com"
}
```

I've labelled the main attribute and start script from the _package.json_ file from the test app of this repo. The default behaviour of sti-nodejs is to do `npm start`. `npm start` executes the start script. By default, nodemon will execute the main attribute of the _package.json_ file if it exists. If the main attribute doesn't appear in the _package.json_ file, it executes the start script. So, in order to achieve some sort of uniform functionality between production and development modes, the user should remove the start script.

I can edit this doc to clarify the purpose and pitfall.